lib/open_project/acts_as_url/adapter/op_active_record.rb

```diff
@@ -57,6 +60,10 @@ def url_attribute(instance)
         def modify_base_url
           super
 
+          if settings.respond_to?(:custom_rule)
+            self.base_url = instance_exec(&settings.custom_rule)
```

Comment:
I dislike `instance_exec` and would prefer the more explicit solution.
If we could please just use a normal lambda here which is passed the base URL as a parameter that would be great. I.e.

```
settings.custom_rule.call base_url
```

Also looking at line 67 `base_url` may, apparently, very well be empty too so we should handle that case too.
That is not call the `custom_rule` lambda in that case (unless we want it to be usable to define a fallback).